src/libs/svg/lv_svg_decoder.c

```diff
@@ -0,0 +1,326 @@
+
+    if(dsc->src_type == LV_IMAGE_SRC_FILE) {
+        const char * fn = dsc->src;
+        if(lv_strcmp(lv_fs_get_ext(fn), "svg") == 0) {              /*Check the extension*/
+
+            svg_data = alloc_file(fn, &svg_data_size);
+            if(svg_data == NULL) {
+                LV_LOG_WARN("can't load file: %s", (const char *)dsc->src);
+                LV_PROFILER_DECODER_END_TAG("lv_svg_decoder_open");
+                return LV_RESULT_INVALID;
+            }
+
+        }
+        else {
+            LV_PROFILER_DECODER_END_TAG("lv_svg_decoder_open");
+            return LV_RESULT_INVALID;
+        }
+    }
+    else if(dsc->src_type == LV_IMAGE_SRC_VARIABLE) {
+        const lv_image_dsc_t * img_dsc = dsc->src;
+        svg_data = (uint8_t *)img_dsc->data;
+        svg_data_size = (uint32_t)img_dsc->data_size;
+    }
+    else {
+        LV_PROFILER_DECODER_END_TAG("lv_svg_decoder_open");
+        return LV_RESULT_INVALID;
+    }
+
+    lv_svg_node_t * svg_doc = lv_svg_load_data((char *)svg_data, svg_data_size);
+    lv_svg_render_obj_t * draw_list = lv_svg_render_create(svg_doc);
+
+    if(dsc->src_type == LV_IMAGE_SRC_FILE) {
+        lv_free(svg_data);
+    }
+    lv_svg_node_delete(svg_doc);
+
+    /* create a fake draw_buf object */
+    lv_draw_buf_t * draw_buf = lv_zalloc(sizeof(lv_draw_buf_t));
+
+    draw_buf->header.w = 1;
+    draw_buf->header.h = 1;
+    draw_buf->header.cf = LV_COLOR_FORMAT_ARGB8888;
+    draw_buf->header.flags = LV_IMAGE_FLAGS_ALLOCATED | LV_IMAGE_FLAGS_CUSTOM_DRAW;
+    draw_buf->header.stride = 4;
+    draw_buf->header.magic = LV_IMAGE_HEADER_MAGIC;
+    draw_buf->data = NULL;
+    draw_buf->unaligned_data = (void *)draw_list;
+    draw_buf->data_size = lv_svg_render_get_size(draw_list);
+    draw_buf->handlers = &_svg_draw_buf_handler;
+
+    dsc->decoded = draw_buf;
+
+    if(!dsc->args.no_cache && lv_image_cache_is_enabled()) {
+
+        lv_image_cache_data_t search_key;
+        search_key.src_type = dsc->src_type;
+        search_key.src = dsc->src;
+        search_key.slot.size = dsc->decoded->data_size;
+
+        lv_cache_entry_t * entry = lv_image_decoder_add_to_cache(decoder, &search_key, draw_buf, NULL);
+
+        if(entry == NULL) {
+            lv_draw_buf_destroy(draw_buf);
+            LV_PROFILER_DECODER_END_TAG("lv_svg_decoder_open");
+            return LV_RESULT_INVALID;
+        }
+        dsc->cache_entry = entry;
+    }
+
+    LV_PROFILER_DECODER_END_TAG("lv_svg_decoder_open");
+    return LV_RESULT_OK;
+}
+
+static void svg_decoder_close(lv_image_decoder_t * decoder, lv_image_decoder_dsc_t * dsc)
+{
+    LV_UNUSED(decoder); /*Unused*/
+
+    if(dsc->args.no_cache ||
+       !lv_image_cache_is_enabled()) lv_draw_buf_destroy((lv_draw_buf_t *)dsc->decoded);
+}
+
+static uint8_t * alloc_file(const char * filename, uint32_t * size)
+{
+    uint8_t * data = NULL;
+    lv_fs_file_t f;
+    uint32_t data_size;
+    uint32_t rn;
+    lv_fs_res_t res;
+
+    *size = 0;
```

Comment:
If size is NULL you might have silent buffer overflow issues, if this parameter is mandatory it would be good to check its valid before proceed.